packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/BroadcastReceiverRegistrar.java

```diff
@@ -0,0 +1,34 @@
+package io.flutter.plugins.connectivity;
+
+import android.content.BroadcastReceiver;
+import androidx.annotation.NonNull;
+import io.flutter.plugin.common.EventChannel;
+
+/** Responsible for constructing a BroadcastReceiver as well as registering and unregistering it. */
+public interface BroadcastReceiverRegistrar {
+
+  /**
+   * Triggered when it is ready for the BroadcastReceiver to be registered. Register the receiver in
```

Comment:
I think this is a little unclear because it's phrased using passive voice, so it's not obvious who is doing what. https://developers.google.com/style/voice

I'd rephrase the first sentence to `<Object> triggers this when it's ready for the BroadcastReceiver to be registered.` With how it is now I think the worry is that you the developer need to trigger it somehow in some other place. I think the second sentence is clear as-is.